pkg/pkg.bzl

```diff
@@ -132,12 +132,16 @@ def _pkg_tar_impl(ctx):
     if ctx.attr.empty_dirs:
         args += ["--empty_dir=%s" % empty_dir for empty_dir in ctx.attr.empty_dirs]
     if ctx.attr.extension and not ctx.executable.compressor:
-        dotPos = ctx.attr.extension.find(".")
-        if dotPos > 0:
-            dotPos += 1
-            args += ["--compression=%s" % ctx.attr.extension[dotPos:]]
-        elif ctx.attr.extension == "tgz":
-            args += ["--compression=gz"]
+        compression = None
+        dotPos = ctx.attr.extension.rfind(".")
+        if dotPos >= 0:
+            compression = ctx.attr.extension[dotPos+1:]
+        else:
+            compression = ctx.attr.extension
+        if compression == "tgz":
+            compression = "gz"
```

Comment:
If you're looking for completeness, you could also do `tbz2` and `txz` for `bzip2` and `xz`-compressed-archives.

Consider also adding `.tbz2` and `txz` to `tar_filetype` at the top of the file if you do this.  Might be worth a ticket if you don't here.

Reply:
I don't understand the landscape of file naming to know if I want completeness.
My personal belief has always been that you have one extension, not two.  That is .tgz is correct and .tar.gz is an abomination. Then I met people with other beliefs. I'ld like to defer adding more until we get requests.